scripts/deploy/src/install.js

```diff
@@ -0,0 +1,136 @@
+import UserRuntimeError from './error.js';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const readFile = function(f) {
+  try {
+    return yaml.load(fs.readFileSync(f, 'utf8'));
+  } catch (err) {
+    console.error(err);
+    process.exit(1);
+  }
+};
+
+const prepare = function(f) {
+  const values = readFile(f);
+  const environment = values.environment || '';
+  const scriptPath = path.join(__dirname, 'prepare.sh');
+  child_process.execSync(`${scriptPath} ${environment}`, { stdio: 'inherit' }); //NoSONAR
+};
+
+const loadValues = function(f) {
+  if (!f) {
+    console.error('No values file provided. Please specify a values file using -f <file>');
+    process.exit(1);
+  }
+  return readFile(f);
+};
+
+const determineNamespace = function(values) {
+  const namespace = values.namespace || '';
+  if (!namespace) {
+    console.error('Namespace is not specified.');
+    process.exit(1);
+  }
+  return namespace;
+};
+
+const get_image_tag = async function(chtversion) {
+  const response = await fetch(`https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:${chtversion}`);
+  const data = await response.json();
+  const tag = data.tags && data.tags.find(tag => tag.container_name === 'cht-api');
+  if (!tag) {
+    return Promise.reject(new UserRuntimeError('cht image tag not found'));
+  }
+  return tag.image.split(':').pop();
+};
+
+const get_chart_version = function(values) {
+  return values.cht_chart_version || DEFAULT_CHART_VERSION;
+};
+
+const helmInstallOrUpdate = function(valuesFile, namespace, values, image_tag) { //NoSONAR
+  const chart_version = get_chart_version(values);
+  ensureMedicHelmRepo();
+  const project_name = values.project_name || '';
+  const namespaceExists = checkNamespaceExists(namespace);
+
+  try {
+    const releaseExists = child_process.execSync(`helm list -n ${namespace}`).toString(); //NoSONAR
+    if (releaseExists.includes(project_name)) { //NoSONAR
+      console.log('Release exists. Performing upgrade.');
+      child_process.execSync(`helm upgrade` + //NoSONAR
+        ` --install ${project_name} ${CHT_CHART_NAME}` + //NoSONAR
+        ` --version ${chart_version} --namespace ${namespace}` + //NoSONAR
+        ` --values ${valuesFile} --set cht_image_tag=${image_tag}`, { stdio: 'inherit' }); //NoSONAR
+      console.log(`Instance at ${values.ingress.host} upgraded successfully.`);
+    } else {
+      console.log('Release does not exist. Performing install.');
+      const createNamespaceFlag = namespaceExists ? '' : '--create-namespace';
+      child_process.execSync(`helm install ${project_name} ${CHT_CHART_NAME}` + //NoSONAR
+        ` --version ${chart_version} --namespace ${namespace} ${createNamespaceFlag}` + //NoSONAR
+        ` --values ${valuesFile} --set cht_image_tag=${image_tag}`, { stdio: 'inherit' }); //NoSONAR
+      console.log(`Instance at ${values.ingress.host} installed successfully.`);
```

Comment:
It's nice to make this a full link:

```suggestion
      console.log(`Instance installed successfully: https://${values.ingress.host}`);
```